### PR TITLE
RDKit first build

### PR DESCRIPTION
**Pull request by sumit0190**
- Forked from `conda-forge` with some modifications to make the recipe simpler and adapt it to our standards.
- For some reason, some Prefect builds randomly fail after building the package as the next "task" fails to start. You can verify through the logs that the package is built and tested correctly.
- `win-64` had to be skipped because of some issues with `boost`. The same recipe works with `boost` from `conda-forge` so it's an issue with our `boost` for sure, but it will take some time to figure it out and `win-64` isn't a priority.
- `s390x` is not supported.
- **THIS PACKAGE WILL NOT BE UPLOADED TO** `defaults`, so once reviewed we will just merge and upload to Snowflake's private channel.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,26 +9,36 @@ package:
...
+  ignore_run_exports:
+    # This is not required by any library, it's just exported by cairo.
+    - pixman
 
 requirements:
   build:
+    - {{ compiler('cxx') }}
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - numpy                                  # [build_platform != target_platform]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Cross-plartform dependencies can be removed

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,26 +9,36 @@ package:
...
-    - {{ compiler('cxx') }}
     - cmake
-    - jom  # [win]
-    - make  # [unix]
+    - jom                                    # [win]
+    - make                                   # [unix]
     - pkg-config
   host:
-    - boost
+    - libboost
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:38]: Do we need `boost-cpp `too? See https://github.com/rdkit/rdkit/blob/Release_2022_09_4/Docs/Book/Install.md#macos-1012-sierra-python-3-environment

File: recipe/build.sh [hunk 3]
```diff
@@ -29,6 +29,8 @@ cmake ${CMAKE_ARGS} \
     -D RDK_BUILD_INCHI_SUPPORT=ON \
     -D RDK_BUILD_FREESASA_SUPPORT=ON \
     -D RDK_BUILD_YAEHMOP_SUPPORT=ON \
+    -D RDK_BUILD_XYZ2MOL_SUPPORT=ON \
+    -D RDK_BUILD_PYTHON_WRAPPERS=ON \
```
Comment (skupr-anaconda) [inline recipe/build.sh:33]: Is there a missing CMage option `PYTHON_INCLUDE_DIR` (for `osx`)? See https://github.com/rdkit/rdkit/blob/Release_2022_09_4/Docs/Book/Install.md#macos-1012-sierra-python-3-environment

File: recipe/meta.yaml [hunk 4]
```diff
@@ -9,26 +9,36 @@ package:
...
   fn: {{ filename }}
   url: https://github.com/rdkit/rdkit/archive/{{ filename }}
-  sha256: 38e6fb9f063b6132310f17e654f2c4350876f9164b0a17b49fe3df7d0555a744
+  sha256: edd30682cc3031cf3f2b1a400f453629db332a1018f355cd3f7ff76b2f7f5398
 
 build:
   number: 0
-  skip: true  # [aarch64]
+  # We skip s390x because it's not supported, and win because of boost weirdness.
+  # win SHOULD work though, and this recipe does build with boost from CF so it's
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Could this recipe be helpful https://github.com/rdkit/conda-rdkit/tree/master/rdkit ?

File: recipe/meta.yaml (hunk 1 above)
Comment (sumit0190) [reply recipe/meta.yaml]: Oops, I forgot about that. Fixed.

File: recipe/meta.yaml (hunk 2 above)
Comment (sumit0190) [reply recipe/meta.yaml:38]: No, our `boost-cpp` is just the same as `boost` and has been added to make porting recipes easier from CF. 
https://github.com/conda-forge/boost-cpp-feedstock/blob/main/recipe/meta.yaml
Our newer recipes should **NOT** use boost_cpp and our aim is to remove it altogether.

File: recipe/build.sh (hunk 3 above)
Comment (sumit0190) [reply recipe/build.sh:33]: This is not needed anymore; if the `find` call in CMake succeeds this variable is set automatically.
https://github.com/rdkit/rdkit/blob/4d4b92765197df1e60893858cc68f0eeb62adf0e/CMakeLists.txt#L276

File: recipe/meta.yaml (hunk 4 above)
Comment (sumit0190) [reply recipe/meta.yaml]: Hmm I already tried it and our recipe is practically the same, so I have no clue what's going on. I will have to investigate it in detail when I get some time. In essence, CMake is unable to find `boost_python` even though it exists in our `libboost` and `py-boost` packages, and curiously it only happens on `win`.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 5]
```diff
@@ -9,26 +9,35 @@ package:
...
+    # This is not required by any library, it's just exported by cairo.
+    - pixman
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
     - {{ compiler('cxx') }}
+    - python
```
Comment (cbouss) [inline recipe/meta.yaml:30]: Is python and numpy really required in build?

**Review by cbouss — APPROVED** [review]
I think this is okay for a first version to a specified channel.

Review notes:
- A few dependencies are downloaded by cmake (avalon, inchi...) at some point we may want to have them un-vendored.
- The pypi wheels only list numpy and pillow as required dependencies. However there are others used in the code (sqlalchemy, matplotlib...). I couldn't file a reference file though.. Following cf sounds fair here.

File: recipe/meta.yaml (hunk 5 above)
Comment (sumit0190) [reply recipe/meta.yaml:30]: Yeah it looks like it.

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**